src/Pulsar.Client/Internal/ConsumerImpl.fs

```diff
@@ -776,45 +777,55 @@ type internal ConsumerImpl internal (consumerConfig: ConsumerConfiguration, clie
 
         member this.AcknowledgeAsync (msgId: MessageId) =
             task {
-                connectionHandler.CheckIfActive()
+                let exn = connectionHandler.CheckIfActive()
+                interceptors.OnAcknowledge(this, msgId, exn)
+                throwIfNotNull exn
+                
                 mb.Post(Acknowledge(msgId, AckType.Individual))
             }
             
         member this.AcknowledgeAsync (msgs: Messages) =
             task {
-                connectionHandler.CheckIfActive()
+                for msg in msgs do
+                    interceptors.OnAcknowledge(this, msg.MessageId, null)
```

Comment:
Why is null there?

Reply:
You suggested removing the check, maybe I misunderstood
https://github.com/RahimovIR/pulsar-client-dotnet/commit/d459f9526240adf4a7764efff31f1b73b8c7b0b9#r37607525